source/FreeRTOS_IP_Utils.c

```diff
@@ -876,12 +879,22 @@ uint16_t usGenerateChecksum( uint16_t usSum,
     }
 
     /* Word (32-bit) aligned, do the most part. */
-    xLastSource.u32ptr = ( xSource.u32ptr + ( uxDataLengthBytes / 4U ) ) - 3U;
+
+    ulSize = ( uintptr_t ) ( ( uxDataLengthBytes / 4U ) * 4U );
```

Comment:
Would size_t be a better type cast for this?

Reply:
i can give it a shot